### ISSUE TITLE
MD027: False positive on typeset poetry

### ISSUE THREAD

**Issue by ssokolow** [markdownlint/markdownlint#510]
## Description
When typesetting a poem in a blockquote using backslashes for explicit line endings, a nonsensical MD027 report is produced.

### Environment
```
% lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.5 LTS
Release:        22.04
Codename:       jammy
```
**MDL Version**
```
% mdl --version
0.13.0
```

### Expected Behavior
No output.

### Actual Behavior
```
../WISHLIST.md:2649: MD027 Multiple spaces after blockquote symbol
```

## Replication Case
```
Here's a placeholder for a typeset poem

> line1 \
> line2 \
> -- Author's Name

Do you see too many spaces here?
```

Problem appears when running `mdl name_of_file.md` or through Vim integration via ALE.

**Comment by nbehrnd** [comment]
@ssokolow markdownlint stumbles over the n-dash leading the author's name you encoded like in LaTeX/tex, restructured text/rst.

A bypass is to write the markdown file in UTF-8 encoding instead of ASCII (which `file my_file.txt` indicates).  The editors I typically use for markdown (vim; sometimes a GUI like geany, or gedit) only change for the later if characters outside the ASCII table are actually used in words like `école`, `garçon`, etc.

To access these

- if rarely used, I copy-paste them from the (special) character map (whisker menu, accessories section).  In the specific case, the n-dash is in the script section of _common_ as `U+2013 EN DASH` where `U+2013` indicates the "UTF-8 address" in question.
- if frequently used, type them directly.  In my setup, the chord of `Ctrl` + `Shift` + `u`  opens a mini menu with a leading lower case `u`.  Release the three keys, the subsequent input of `2013` is confirmed by `ENTER` to yield the n-dash.  Compare ASCII minus/hyphen/dash `-` vs UTF-8 n-dash `–` (in some [non-proportional] fonts, they are difficult to discern).  The approach works well in vim, gedit, geany; Abiword; now here in Firefox, etc.

The following will not cause an issue to mdl (version 0.13.0)

```
Here's a placeholder for a typeset poem

> line1 \
> line2 \
> – Author's Name

Do you see too many spaces here?
```

It depends on the overall workflow if leaving ASCII for UTF-8 is an advantage.*  If you need to return to ASCII, `iconv` in Linux (and the BASH shell within the couple of git/[tortoise git](https://tortoisegit.org/) for Windows) offers you a conversion, for instance

```shell
iconv -f UTF-8 -t ASCII/TRANSLIT input_file.md > output_file.md
```

\* Anecdotally, I recently tested three workflows using the same Pandoc flavored markdown file as source ([link](https://github.com/nbehrnd/bader_article)) to eventually render a pdf file.  Contrasting to the route via pdfLaTeX or groff, n-dashes expressed with the single `U+2013` character were incompatible to [rst2pdf](https://rst2pdf.org/), and thus I had to restrain the character set/split single character `U+2013` into `--` in the source.

**Comment by ssokolow** [comment]
> A bypass is to ...

Thanks for the effort, but I'm reporting this bug for the benefit of others.

I already added `~MD027` to the growing `rules` list in my `~/.mdlrc` and went on with my day.

(Granted, half of the disabled rules are there because I have reformat-on-save, Prettier is "opinionated", and it's not worth the bother to check if markdownlint can be configured to match what Prettier is going to ensure anyway, but the point remains.)

As for typing UTF-8, I've had enough bad experiences with mojibake over the years that I implemented shortcodes in my pulldown-cmark frontend specifically so the authoritative copies of my documents could contain ASCII-only representations of things I *can* type using my compose key (eg. ±) without me having to memories HTML entities.

**Comment by DavidAnson** [comment]
FYI, here are some tips from the Node version that may be helpful for working with Prettier: https://github.com/DavidAnson/markdownlint/blob/main/doc/Prettier.md

**Comment by ssokolow** [comment]
Thanks. That could come in handy if I want something less strict than `prettier --check` in my CI runs.

**Comment by jamesquilty** [comment]
I've just been bitten by the bug underlying this report, my false positive is on blockquotes beginning with an ellipsis (three dots):

```text
> ... some text.
```

A guess, based on a cursory reading of commit 3d760ba, is that `extract_as_text()` is stripping non-text characters, such as punctuation, resulting in `>  some text.` and a false positive of MD027.
